src/Reflex/Class.hs

```diff
@@ -679,15 +678,12 @@ instance (Reflex t, IsString a) => IsString (Behavior t a) where
 
 instance Reflex t => Monad (Behavior t) where
   a >>= f = pull $ sample a >>= sample . f
-  -- Note: it is tempting to write (_ >> b = b); however, this would result in (fail x >> return y) succeeding (returning y), which violates the law that (a >> b = a >>= \_ -> b), since the implementation of (>>=) above actually will fail.  Since we can't examine 'Behavior's other than by using sample, I don't think it's possible to write (>>) to be more efficient than the (>>=) above.
```

Comment:
@ryantrinkle is this obsolete now that MonadFail got pulled out?

Reply:
Yeah, we could use the simpler implementation now.